### PR TITLE
Fix tempfilter with IEnumerable

### PR DESCRIPTION
**Pull request by and-rewsmith**
TempFilter tests sometimes fail due to the modules starting before the necessary edgeHub routes are in place. I made a fix for this where the new e2e framework deploys in configurations. First it deploys edgeAgent / edgeHub, then the rest of the modules.

### CODE REVIEW

File: test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs [hunk 1]
```diff
@@ -84,9 +88,15 @@ public EdgeConfiguration Build()
                         ["properties.desired"] = module.DesiredProperties
                     };
                 }
+
+                if (module.Name == "$edgeHub")
```
Comment (philipktlin) [inline test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs]: use string.Equals with comparison option

File: test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs [hunk 2]
```diff
@@ -72,7 +76,7 @@ public EdgeConfiguration Build()
             var moduleNames = new List<string>();
             var moduleImages = new List<string>();
 
-            foreach (ModuleConfiguration module in modules)
+            foreach (ModuleConfiguration module in modules.OrderBy(m => m.Name != "$edgeHub"))
```
Comment (philipktlin) [inline test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs]: should it be better to handle edgeHub first and use where clause for filtering to handle other modules

File: test/Microsoft.Azure.Devices.Edge.Test.Common/EdgeRuntime.cs [hunk 3]
```diff
@@ -48,13 +48,18 @@ public async Task<EdgeDeployment> DeployConfigurationAsync(Action<EdgeConfigBuil
             addConfig(builder);
 
             DateTime deployTime = DateTime.Now;
-            EdgeConfiguration config = builder.Build();
+            IEnumerable<EdgeConfiguration> configs = builder.BuildConfigurationStages();
```
Comment (and-rewsmith) [inline test/Microsoft.Azure.Devices.Edge.Test.Common/EdgeRuntime.cs]: Let me know what you think of this name change

File: test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs (hunk 2 above)
Comment (and-rewsmith) [reply test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs]: I made this change

File: test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs (hunk 1 above)
Comment (and-rewsmith) [reply test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs]: I made this change

File: test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs [hunk 4]
```diff
@@ -72,21 +75,25 @@ public EdgeConfiguration Build()
             var moduleNames = new List<string>();
             var moduleImages = new List<string>();
 
-            foreach (ModuleConfiguration module in modules)
+            string edgeHubModuleId = "$edgeHub";
```
Comment (philipktlin) [inline test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs]: uses Constants.EdgeHubModuleIdentityName in EdgeAgent.Core project

File: test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs [hunk 5]
```diff
@@ -167,6 +171,25 @@ ModuleConfiguration BuildEdgeAgent(IEnumerable<ModuleConfiguration> configs)
             return agentBuilder.Build();
         }
 
+        bool IsSystemModule(ModuleConfiguration module)
+        {
+            return module.Name.Equals("$edgeHub") || module.Name.Equals("$edgeAgent");
```
Comment (philipktlin) [inline test/Microsoft.Azure.Devices.Edge.Test.Common/config/EdgeConfigBuilder.cs]: Check with Damon if he have any concerns to reference EdgeAgent.Core project for these constants.

**Pull request merged**